### PR TITLE
upgrade logrus to v1.4.2

### PR DESCRIPTION
**Pull request by thinkerou**
# Description

- upgrade logrus to v1.4.2
- remove `depguard` from golintci.yml
- use `go test ./...` instead of `go test ./... -mod=vendor`

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -51,12 +51,12 @@ endif
 ################################################################################
 .PHONY: test
 test:
-	go test ./... -mod=vendor
+	go test ./...
```
Comment (youngbupark) [inline Makefile]: we need to keep -mod=vendor. can you run `go mod vendor` in your change ?

cc/ @yaron2

Comment (thinkerou) [reply Makefile]: @youngbupark Yes.
we use `go mod` and `GOPROXY`, so I suggest we not use `vendor`.

Comment (youngbupark) [reply Makefile]: > @youngbupark Yes.
> we use `go mod` and `GOPROXY`, so I suggest we not use `vendor`.

@thinkerou Sorry for the late reply. This pr is to upgrade logurus to 1.4.2, which is not related to remove vendor.  So, let's keep using vendor now. Instead, please create the issue to remove vendor option in dapr/dapr.

cc/ @yaron2

Comment (thinkerou) [reply Makefile]: @youngbupark OK, thanks!

**Comment by thinkerou** [comment]
closing the pr.

**Pull request closed without merging**